src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._comm = clientComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._serverObject = clientComm:BuildObject()
+	self._owner = Players:GetPlayerFromCharacter(character)
+	self.Instance = character
+
+	self._trove:Connect(character.AncestryChanged, function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
+	if not self:IsOwner() then
+		return self
+	end
+
+	ContextActionService:BindAction("Sprint",
+		function(actionName: string, userInputState: Enum.UserInputState)
+			local humanoid = character:FindFirstChildWhichIsA("Humanoid")
+
+			if userInputState == Enum.UserInputState.Begin and humanoid then
+				self:StartSprinting()
+			elseif userInputState == Enum.UserInputState.End then
+				self:StopSprinting()
+			end
+
+			return Enum.ContextActionResult.Pass
+		end,
+	false, Enum.KeyCode.LeftShift)
```

Comment:
Context Action Service being ugly as usual. There's not really a cure for it, at least yet. We should definitely look into a better API wrapper for it.

Reply:
Yeah, I mentioned in in a previous review, we should eventually have some sort of proper input binds system, even if it's not exposed to players as something they can configure themselves.